modules/globebrowsing/util/globelabelscomponent.cpp

```diff
@@ -0,0 +1,842 @@
+                }
+
+                _font = font;
+                initializeFonts();
+            }
+        }
+    }
+}
+
+bool GlobeLabelsComponent::initializeGL() {
+    return true;
+}
+
+void GlobeLabelsComponent::initializeFonts() {
+    if (_font == nullptr) {
+        _font = openspace::global::fontManager.font(
+            "Mono",
+            static_cast<float>(_labelsFontSize),
+            ghoul::fontrendering::FontManager::Outline::Yes,
+            ghoul::fontrendering::FontManager::LoadGlyphs::No
+        );
+    }
+}
+
+
+bool GlobeLabelsComponent::deinitialize() {
+    return true;
+}
+
+bool GlobeLabelsComponent::isReady() const {
+    return true;
+}
+
+void GlobeLabelsComponent::update() {
+        
+}
+
+bool GlobeLabelsComponent::loadLabelsData(const std::string& file) {
+    bool success = true;
+    if (_labelsDataPresent) {
+        std::string cachedFile = FileSys.cacheManager()->cachedFilename(
+            ghoul::filesystem::File(file),
+            "GlobeLabelsComponent|" + identifier(),
+            ghoul::filesystem::CacheManager::Persistent::Yes
+        );
+
+        bool hasCachedFile = FileSys.fileExists(cachedFile);
+        if (hasCachedFile) {
+            LINFO(fmt::format(
+                "Cached file '{}' used for labels file '{}'",
+                cachedFile,
+                file
+            ));
+
+            success = loadCachedFile(cachedFile);
+            if (success) {
+                return true;
+            }
+            else {
+                FileSys.cacheManager()->removeCacheFile(file);
+                // Intentional fall-through to the 'else' to generate the cache
+                // file for the next run
+            }
+        }
+        else {
+            LINFO(fmt::format("Cache for labels file '{}' not found", file));
+        }
+        LINFO(fmt::format("Loading labels file '{}'", file));
+
+        success = readLabelsFile(file);
+        if (!success) {
+            return false;
+        }
+
+        success &= saveCachedFile(cachedFile);
+    }
+    return success;
+}
+
+bool GlobeLabelsComponent::readLabelsFile(const std::string& file) {
+    try {
+        std::fstream csvLabelFile(file);
+        if (!csvLabelFile.good()) {
+            LERROR(fmt::format("Failed to open labels file '{}'", file));
+            return false;
+        }
+        if (csvLabelFile.is_open()) {
+            char line[4096];
+            _labels.labelsArray.clear();
+            while (!csvLabelFile.eof()) {
+                csvLabelFile.getline(line, 4090);
+                if (strnlen(line, 4090) > 10) {
+                    LabelEntry lEntry;
+                    char *token = strtok(line, ",");
+                    // First line is just the Header
+                    if (strcmp(token, "Feature_Name") == 0) {
+                        continue;
+                    }
+                    strncpy(lEntry.feature, token, 256);
+                    // Removing non ASCII characters:
```

Comment:
Good that there is a comment about this. Can it be elaborated a little bit? Like, why are non-ascii chars causing issues, and in which circumstances do they exist in input files

Reply:
Also consider sanitizing the string in a separate function, to make the code more concise.

Reply:
Non-ascii characters aren't displayed correctly by the text rendering (We don't have the non-ascii character in the texture atlas if I am not wrong).

Reply:
Ok, please add that as a comment in the code, so the strncpy can be removed once we have non-ascii rendering in place.

Reply:
Done